### ISSUE TITLE
Investigate disk i/o utilization not stable

### ISSUE THREAD

**Issue by oiooj** [influxdata/influxdb#9209]
#9204 Fixed higher disk i/o utilization, and it works for me.  Thanks a lot. But the disk i/o utilization not stable.

After 9024 fix:
![image](https://user-images.githubusercontent.com/2253095/33749590-83278484-dc09-11e7-8720-cd9049e8aa45.png)

**Comment by jwilder** [comment]
@oiooj What is your series cardinality and what do you have set for `cache-snapshot-memory-size`?

**Comment by oiooj** [comment]
@jwilder `series num: 1488994` , `db num: 913`. I don't change the `cache-snapshot-memory-size` from 1.2.4 :

```
  # CacheSnapshotMemorySize is the size at which the engine will
  # snapshot the cache and write it to a TSM file, freeing up memory
  # cache-snapshot-memory-size = 26214400
```

**Comment by oiooj** [comment]
Upate : after #9204 fixed:

![image](https://user-images.githubusercontent.com/2253095/33802280-afdbe690-ddae-11e7-8560-63443ac52c5c.png)

but the server memory  ran out (OOM risk),  so I have to roll back. 😢

**Comment by szibis** [comment]
After this weekend of using this fix i have twice OOM on influxDB, just like @oiooj suggest big bump on memory usage. Latest here:
```
Dec 11 12:35:00 ip-172-23-49-106 kernel: [4156136.995123] Out of memory: Kill process 91106 (influxd) score 938 or sacrifice child
```

**Comment by szibis** [comment]
![image](https://user-images.githubusercontent.com/329831/33836558-00651876-de8a-11e7-8983-6d7b78705565.png)

**Comment by jwilder** [comment]
@szibis @oiooj Can you guys try setting

```
cache-snapshot-memory-size = "25m"
```

and see if that resolves this?

@szibis How many databases do you have?

**Comment by jwilder** [comment]
Also, can you attach the stack trace printed by the OOM killer?

**Comment by szibis** [comment]
@jwilderI have 5-6 databases and only two big there (diamond and telegraf aka infra)

There is OOM in attachment.

[telegraf_influxdb_killed.txt](https://github.com/influxdata/influxdb/files/1551219/telegraf_influxdb_killed.txt)

I just changed to ```cache-snapshot-memory-size = "25m"``` on one node.

**Comment by oiooj** [comment]
![image](https://user-images.githubusercontent.com/2253095/34089409-c446ccde-e3ea-11e7-9a61-4b761c121b04.png)

The mem is OK now.

**Comment by jwilder** [comment]
Fixed in 1.5
